### PR TITLE
refactor: getrusage wrapper to common

### PR DESCRIPTION
**Pull request by coryan**
De-duplicate the wrapper to measure CPU usage via getrusage(), we are
going to need a third copy for Pub/Sub.

Part of the work for #4565

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5334)

### CODE REVIEW

File: google/cloud/testing_util/BUILD [hunk 1]
```diff
@@ -18,10 +18,36 @@ licenses(["notice"])  # Apache 2.0
...
 cc_library(
     name = "google_cloud_cpp_testing",
     srcs = google_cloud_cpp_testing_srcs,
     hdrs = google_cloud_cpp_testing_hdrs,
+    defines = select({
+        ":linux_x86_64": [
+            "GOOGLE_CLOUD_CPP_HAVE_GETRUSAGE=1",
+            "GOOGLE_CLOUD_CPP_HAVE_GETRUSAGE_THREAD=1",
+        ],
+        ":windows": [],
```
Comment (devbww) [inline google/cloud/testing_util/BUILD:48]: Does "macos" deserve an entry here, or does "windows" not deserve one?

File: google/cloud/testing_util/cpu_usage.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  void Stop();
+
+  //@{
+  /**
+   * @name Measurement results.
+   *
+   * @note The values are only valid after calling Start() and Stop().
+   */
+  std::chrono::microseconds elapsed_time() const { return elapsed_time_; }
+  std::chrono::microseconds cpu_time() const { return cpu_time_; }
```
Comment (devbww) [inline google/cloud/testing_util/cpu_usage.h:46]: Nit: It is a little strange for this to be called `CpuUsage` when it supports both `cpu_time()` and `elapsed_time()`.  Perhaps something more like `Timer` would be clearer?

**Review by devbww — APPROVED** [review]

File: google/cloud/testing_util/BUILD (hunk 1 above)
Comment (coryan) [reply google/cloud/testing_util/BUILD]: Fixed, thanks.

File: google/cloud/testing_util/cpu_usage.h (hunk 2 above)
Comment (coryan) [reply google/cloud/testing_util/cpu_usage.h]: Fixed.

**Comment by coryan** [comment]
PTAL

**Review by devbww — APPROVED** [review]

**Pull request merged**